core/src/main/java/io/confluent/rest/Application.java

```diff
@@ -136,16 +140,32 @@ public Application(T config, String path, String listenerName) {
         this.getMetricsTags(),
         config.getString(RestConfig.METRICS_JMX_PREFIX_CONFIG));
 
-    if (customRequestLog == null) {
+    if (inputRequestLog == null) {
       Slf4jRequestLogWriter logWriter = new Slf4jRequestLogWriter();
       logWriter.setLoggerName(config.getString(RestConfig.REQUEST_LOGGER_NAME_CONFIG));
       // %{ms}T logs request time in milliseconds
       requestLog = new CustomRequestLog(logWriter, requestLogFormat());
     } else {
-      requestLog = customRequestLog;
+      requestLog = inputRequestLog;
     }
   }
 
+  /**
+   * Set DosFilter.listeners for global-dosfilter
+   */
+  public void setGlobalDosfilterListeners(
```

Comment:
Not sure we need this public api as those listeners are only triggered on DoSFilter, for which rest-utils is the only repo that uses dosfilter.

Reply:
I chose the option to offer this as a public API as user of `rest-utils` would want to do custom-actions on when each of those filters trigger. For example, `kafka-rest` will use it to configure dosfilter.listeners that add request-attributes for when requests get-ratelimited by Jetty filters(see this [code](https://github.com/confluentinc/kafka-rest/blob/1cedf144e133223ab1e60c630ee2014661a0f87b/kafka-rest/src/main/java/io/confluent/kafkarest/KafkaRestMain.java#L57) WIP PR). These request-attributes are then logged in the custom logger.